sensor/common/clusterentities/store.go

```diff
@@ -31,6 +36,42 @@ type PublicIPsListener interface {
 	OnRemoved(ip net.IPAddress)
 }
 
+func newEntityStatus(numTicks uint16) *entityStatus {
+	return &entityStatus{
+		ticksLeft:    numTicks,
+		isHistorical: false,
+	}
+}
+
+type entityStatus struct {
+	ticksLeft    uint16
+	isHistorical bool
+}
+
+// markHistorical is called when entity would be deleted.
+// Istead, we mark it as historcal and keep it as long as ticksLeft
```

Comment:
```
// markHistorical is called when entity no longer exists in the cluster.
// Instead of deleting it from memory, we mark it as historical and keep it as long as ticksLeft
```